src/git-hosting-api/git-hosting-api-factory.ts

```diff
@@ -0,0 +1,7 @@
+import { GitHostingAPI, Options } from "./git-hosting-api";
+import GithubAPI from "./github-api";
+import GithubEnterpriseAPI from "./github-enterprise-api";
+
+export function createGitHostingAPI(options: Options): GitHostingAPI {
+  return options.gitHostingServerURL ? new GithubEnterpriseAPI(options) : new GithubAPI(options);
+}
```

Comment:
This is a factory responsible for which `GitHostingAPI` instance should be created according to the configuration.

When an instance of another service is created, (`BitbucketAPI`, `GitlabAPI`, etc...) this factory will be responsible and return the appropriate instance according to the options.